python/tvm/relay/prelude.py

```diff
@@ -533,3 +540,8 @@ def __init__(self, mod):
         self.define_id()
         self.define_compose()
         self.define_iterate()
+
+def tag2constructor(tag):
```

Comment:
I like having the mapping of tags to constructors, but I am not entirely sure about this function. It creates a new module and prelude for that module, so if you are trying to compare constructors with a module that already exists, the pointers may not match. Could it make more sense to have such a function take in a prelude?

(Also, total nitpick, I am not fond of the name since it seems to clash with the style used for other names)

Reply:
@slyubomirsky Due to Jared's comment, I removed the mapping. Instead added an optional field in `ConstructorValueNode` to track the original constructor.